aiida/orm/data/array/trajectory.py

```diff
@@ -50,30 +50,36 @@ def _internal_validate(self, stepids, cells, symbols, positions, times, velociti
         """
         import numpy
 
-        if not isinstance(stepids, numpy.ndarray) or stepids.dtype != int:
-            raise TypeError("TrajectoryData.stepids must be a numpy array of integers")
-        if not isinstance(cells, numpy.ndarray) or cells.dtype != float:
-            raise TypeError("TrajectoryData.cells must be a numpy array of floats")
-        if not isinstance(symbols, numpy.ndarray):
-            raise TypeError("TrajectoryData.symbols must be a numpy array")
+        if not isinstance(symbols, list):
+            raise TypeError("TrajectoryData.symbols must be of type list")
         if any([not isinstance(i, six.string_types) for i in symbols]):
-            raise TypeError("TrajectoryData.symbols must be a numpy array of strings")
+            raise TypeError("TrajectoryData.symbols must be a list of strings")
         if not isinstance(positions, numpy.ndarray) or positions.dtype != float:
             raise TypeError("TrajectoryData.positions must be a numpy array of floats")
+        if stepids is not None:
+            if not isinstance(stepids, numpy.ndarray) or stepids.dtype != int:
+                raise TypeError("TrajectoryData.stepids must be a numpy array of integers")
+        if cells is not None:
+            if not isinstance(cells, numpy.ndarray) or cells.dtype != float:
+                raise TypeError("TrajectoryData.cells must be a numpy array of floats")
         if times is not None:
             if not isinstance(times, numpy.ndarray) or times.dtype != float:
                 raise TypeError("TrajectoryData.times must be a numpy array of floats")
         if velocities is not None:
             if not isinstance(velocities, numpy.ndarray) or velocities.dtype != float:
                 raise TypeError("TrajectoryData.velocities must be a numpy array of floats, or None")
 
-        numsteps = stepids.size
-        if stepids.shape != (numsteps,):
-            raise ValueError("TrajectoryData.stepids must be a 1d array")
-        if cells.shape != (numsteps, 3, 3):
-            raise ValueError("TrajectoryData.cells must have shape (s,3,3), " "with s=number of steps")
-        numatoms = symbols.size
-        if symbols.shape != (numatoms,):
+        if stepids is not None:
+            numsteps = stepids.size
+            if stepids.shape != (numsteps,):
+                raise ValueError("TrajectoryData.stepids must be a 1d array")
+        else:
+            numsteps = positions.shape[0]
+        if cells is not None:
+            if cells.shape != (numsteps, 3, 3):
+                raise ValueError("TrajectoryData.cells must have shape (s,3,3), " "with s=number of steps")
+        numatoms = len(symbols)
+        if numpy.array(symbols).shape != (numatoms,):
```

Comment:
Probably we don't need to go via a numpy array just to get the shape

Reply:
Adjusted in 467d37b42022f6248bc7056bf6f55992a2a7fce6. I removed this test completely b/c there is already this check:
https://github.com/aiidateam/aiida_core/blob/467d37b42022f6248bc7056bf6f55992a2a7fce6/aiida/orm/data/array/trajectory.py#L56-L57
which would trigger immediately if if the list is not 1D. So, this test seems to be superfluous b/c we actually never reach it.